File: MCP服务器/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/insights.py
# region imports
from AlgorithmImports import *
# endregion


class InsightTestAlgorithm(QCAlgorithm):

    def initialize(self):
        self.set_start_date(2024, 1, 8)
        self.set_end_date(2024, 4, 1)
        self.set_cash(100000)
        self.add_equity("SPY", Resolution.DAILY)
        self.add_alpha(
            ConstantAlphaModel(
                InsightType.PRICE, InsightDirection.UP, timedelta(30), 
                0.1, 0.2, 0.3
            )
        )
        self.set_portfolio_construction(EqualWeightingPortfolioConstructionModel())

File: MCP服务器/finance/QuantConnect_mcp-server_3a656b0_python/tests/algorithms/Insights.cs
#region imports
    using System;
    using QuantConnect;
    using QuantConnect.Algorithm.Framework.Alphas;
    using QuantConnect.Algorithm.Framework.Portfolio;
    using QuantConnect.Algorithm;
#endregion
public class InsightTestAlgorithm : QCAlgorithm
{
    public override void Initialize()
    {
        SetStartDate(2024, 1, 8);
        SetEndDate(2024, 4, 1);
        SetCash(100000);
        AddEquity("SPY", Resolution.Daily);
        AddAlpha(
            new ConstantAlphaModel(
                InsightType.Price, InsightDirection.Up, TimeSpan.FromDays(30), 0.1, 0.2, 0.3
            )
        );
        SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
    }
}
